Update nf-azure: Add Azure Compute Gallery image and verification options (#7338)
nextflow-io/nextflow: docs/reference/config.mdx

The client ID for an Azure [managed identity](https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/overview) that is available on all Azure Batch node pools. This identity is used by Fusion to authenticate to Azure storage. If set to `'auto'`, Fusion will use the first available managed identity.

> ##### `azure.batch.pools.<name>.allowUnverifiedImages`

Author: I was thinking about that, but I don't know in which version this PR will be merged, so I found it a bit hard to include that here already

I assume that this tag would be added by whomever eventually merges the PR?

Of course, I can add it if you'd like

The `startTask` that is executed as the node joins the Azure Batch node pool.

##### `azure.batch.pools.<name>.verification`

> The image verification type to match when resolving the VM image from the Batch supported-images list. Can be `verified`, `unverified`, or `any` (default`verified`). Ignored when `virtualMachineImageId` is set.

Reviewer: ```suggestion
The image verification type to match when resolving the VM image from the Batch supported-images list. One of `verified`, `unverified`, or `any` (default `verified`). Ignored when `virtualMachineImageId` is set.
```

Author: No longer relevant

##### `azure.batch.pools.<name>.virtualMachineImageId`

> The resource ID of a custom VM image from an Azure Compute Gallery to use for the pool nodes (e.g. `/subscriptions/<id>/resourceGroups/<group>/providers/Microsoft.Compute/galleries/<gallery>/images/<definition>/versions/<version>`). When set, `publisher` and `offer` are ignored, and `sku` must be set to the Batch node agent SKU id that matches the image OS (e.g. `batch.node.ubuntu 24.04`).

Reviewer: ```suggestion
The resource ID of a custom VM image from an Azure Compute Gallery for the pool nodes (e.g., `/subscriptions/<id>/resourceGroups/<group>/providers/Microsoft.Compute/galleries/<gallery>/images/<definition>/versions/<version>`). When set, `publisher` and `offer` are ignored, and `sku` must be the Batch node agent SKU id matching the image OS (for example, `batch.node.ubuntu 24.04`).

```
